src/deferred.js

```diff
@@ -316,14 +314,17 @@ jQuery.extend( {
 					if ( values === progressValues ) {
 						master.notifyWith( contexts, values );
 					} else if ( !( --remaining ) ) {
-						master.resolveWith( contexts, values );
+						master.resolveWith(
+							contexts.length === 1 ? contexts[ 0 ] : contexts,
+							values
```

Comment:
An array of values passed to `resolveWith` will get spread by apply, unlike with contexts. It does the same thing whether you pass an array of length 1 or just the value.